parser-typechecker/src/Unison/Codebase/SqliteCodebase/Migrations.hs

```diff
@@ -106,17 +106,20 @@ ensureCodebaseIsUpToDate localOrRemote root getDeclType termBuffer declBuffer sh
 
     Region.displayConsoleRegions do
       (`UnliftIO.finally` finalizeRegion) do
+        let migs = migrations getDeclType termBuffer declBuffer root
+        -- The highest schema that this ucm knows how to migrate to.
+        let highestKnownSchemaVersion = fst . head $ Map.toDescList migs
+        currentSchemaVersion <- Sqlite.runTransaction conn Q.schemaVersion
+        when (currentSchemaVersion > highestKnownSchemaVersion) $ UnliftIO.throwIO $ OpenCodebaseUnknownSchemaVersion (fromIntegral currentSchemaVersion)
+        backupCodebaseIfNecessary backupStrategy localOrRemote conn currentSchemaVersion highestKnownSchemaVersion root
+        when shouldPrompt do
+          putStrLn "Press <enter> to start the migration once all other ucm processes are shutdown..."
+          void $ liftIO getLine
         ranMigrations <-
           Sqlite.runWriteTransaction conn \run -> do
-            schemaVersion <- run Q.schemaVersion
-            let migs = migrations getDeclType termBuffer declBuffer root
-            -- The highest schema that this ucm knows how to migrate to.
-            let currentSchemaVersion = fst . head $ Map.toDescList migs
```

Comment:
In the old version we named the latest version we knew about `currentSchemaVersion` which I found pretty confusing, so I tried to clarify this as `currentSchemaVersion` for the current DB version and `highestKnownSchemaVersion`;

So when reviewing, note that the old `currentSchemaVersion` is now `highestKnownSchemaVersion` and the old `schemaVersion` is now `currentSchemaVersion`, sorry for the confusion 😬